### PR TITLE
[WFLY-16755] Update javax.* uses in TX docs. Use JTA 4.0 javadoc links

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.redhat.com/browse/WFLY-16755 (JTA docs part)

@mmusgrov FYI

### CODE REVIEW

**Comment by bstansberry** [comment]
I cancelled TeamCity CI for this to save CI resources as CI doesn't do any useful testing of docs-only changes.

**Comment by rhusar** [comment]
> I cancelled TeamCity CI for this to save CI resources as CI doesn't do any useful testing of docs-only changes.

Always remember there is https://issues.redhat.com/browse/WFCI-51 :-)

**Comment by rhusar** [comment]
Also, needless to say, they weren't actually cancelled and they all ran :-)

File: docs/src/main/asciidoc/_developer-guide/Jakarta_Transactions_Reference.adoc [hunk 1]
```diff
@@ -83,9 +83,9 @@ public class TestServlet extends HttpServlet {
 }
 ----
 
-https://javaee.github.io/javaee-spec/javadocs/javax/transaction/TransactionManager.html[`javax.transaction.TransactionManager`]
+https://jakarta.ee/specifications/transactions/2.0/apidocs/jakarta/transaction/transactionmanagerl[`jakarta.transaction.TransactionManager`]
```
Comment (mmusgrov) [inline docs/src/main/asciidoc/_developer-guide/Jakarta_Transactions_Reference.adoc:86]: typo

**Review by mmusgrov — APPROVED** [review]

File: docs/src/main/asciidoc/_developer-guide/Jakarta_Transactions_Reference.adoc (hunk 1 above)
Comment (bstansberry) [reply docs/src/main/asciidoc/_developer-guide/Jakarta_Transactions_Reference.adoc]: Thanks, @mmusgrov -- I've fixed that.

**Pull request merged**